dali/pipeline/proto/dali.proto

```diff
@@ -81,7 +81,13 @@ message Checkpoint {
     optional bytes operator_state = 2;
   }
 
+  message PipelineCheckpoint {
```

Comment:
Just having second thought about the name, maybe it should be more like ExternalContextCheckpoint? (We have pipeline in cpp backend too). I don't have strong feelings here though, what do you think @klecki?

Reply:
I agree  with @stiepan. I know this is maybe a bit of nitpicking, but I feel like it's more about ExternalSource or Python state.

Reply:
`ExternalContextCheckpoint` sounds good, will change